src/measure_theory/function/locally_integrable.lean

```diff
@@ -0,0 +1,183 @@
+
+This file contains properties of locally integrable functions and of integrability results
+on compact sets.
+
+## Main statements
+
+* `continuous.locally_integrable`: A continuous function is locally integrable.
+
+-/
+
+open measure_theory measure_theory.measure set function topological_space
+open_locale topological_space interval
+
+variables {X Y E : Type*} [measurable_space X] [topological_space X]
+variables [measurable_space Y] [topological_space Y]
+variables [normed_group E] [measurable_space E] {f : X → E} {μ : measure X}
+
+namespace measure_theory
+
+/-- A function `f : X → E` is locally integrable if it is integrable on all compact sets.
+  See `measure_theory.locally_integrable_iff` for the justification of this name. -/
+def locally_integrable (f : X → E) (μ : measure X . volume_tac) : Prop :=
+∀ ⦃K⦄, is_compact K → integrable_on f K μ
+
+lemma integrable.locally_integrable (hf : integrable f μ) : locally_integrable f μ :=
+λ K hK, hf.integrable_on
+
+lemma locally_integrable.ae_measurable [sigma_compact_space X] (hf : locally_integrable f μ) :
+  ae_measurable f μ :=
+begin
+  rw [← @restrict_univ _ _ μ, ← Union_compact_covering, ae_measurable_Union_iff],
+  exact λ i, (hf $ is_compact_compact_covering X i).ae_measurable
+end
+
+lemma locally_integrable_iff [locally_compact_space X] :
+  locally_integrable f μ ↔ ∀ x : X, ∃ U ∈ 𝓝 x, integrable_on f U μ :=
+begin
+  refine ⟨λ hf x, _, λ hf K hK, _⟩,
+  { obtain ⟨K, hK, h2K⟩ := exists_compact_mem_nhds x, exact ⟨K, h2K, hf hK⟩ },
+  { refine is_compact.induction_on hK integrable_on_empty (λ s t hst h, h.mono_set hst)
+      (λ s t hs ht, integrable_on_union.mpr ⟨hs, ht⟩) (λ x hx, _),
+    obtain ⟨K, hK, h2K⟩ := hf x, exact ⟨K, nhds_within_le_nhds hK, h2K⟩ }
+end
+
+section real
+variables [opens_measurable_space X] {A K : set X} {g g' : X → ℝ}
+
+lemma integrable_on.mul_continuous_on_of_subset
+  (hg : integrable_on g A μ) (hg' : continuous_on g' K)
+  (hA : measurable_set A) (hK : is_compact K) (hAK : A ⊆ K) :
+  integrable_on (λ x, g x * g' x) A μ :=
+begin
+  rcases is_compact.exists_bound_of_continuous_on hK hg' with ⟨C, hC⟩,
+  rw [integrable_on, ← mem_ℒp_one_iff_integrable] at hg ⊢,
+  have : ∀ᵐ x ∂(μ.restrict A), ∥g x * g' x∥ ≤ C * ∥g x∥,
+  { filter_upwards [ae_restrict_mem hA] with x hx,
+    rw [real.norm_eq_abs, abs_mul, mul_comm, real.norm_eq_abs],
+    apply mul_le_mul_of_nonneg_right (hC x (hAK hx)) (abs_nonneg _), },
+  exact mem_ℒp.of_le_mul hg (hg.ae_measurable.mul ((hg'.mono hAK).ae_measurable hA)) this,
+end
+
+lemma integrable_on.mul_continuous_on [t2_space X]
+  (hg : integrable_on g K μ) (hg' : continuous_on g' K) (hK : is_compact K) :
+  integrable_on (λ x, g x * g' x) K μ :=
+hg.mul_continuous_on_of_subset hg' hK.measurable_set hK (subset.refl _)
+
+lemma integrable_on.continuous_on_mul_of_subset
+  (hg : integrable_on g A μ) (hg' : continuous_on g' K)
+  (hA : measurable_set A) (hK : is_compact K) (hAK : A ⊆ K) :
+  integrable_on (λ x, g x * g' x) A μ :=
```

Comment:
this lemma is identical to `integrable_on.mul_continuous_on_of_subset`. I think you meant to swap which function is `integrable_on` and which is `continuous_on`. And if that is indeed the intent, would `continuous_on.mul_integrable_on_of_subset` be a better name?

Reply:
I didn't write these lemmas just move them. But I think you're right. @sgouezel is this correct?

Reply:
I think I like the old name better, since the conclusion is also `integrable_on`.